include/aws/io/event_loop.h

```diff
@@ -387,59 +96,12 @@ AWS_IO_API
 int aws_event_loop_current_clock_time(struct aws_event_loop *event_loop, uint64_t *time_nanos);
 
 /**
- * Creates an event loop group, with clock, number of loops to manage, and the function to call for creating a new
- * event loop.
+ * Creation function for event loop groups.
  */
 AWS_IO_API
 struct aws_event_loop_group *aws_event_loop_group_new(
-    struct aws_allocator *alloc,
-    aws_io_clock_fn *clock,
-    uint16_t el_count,
-    aws_new_event_loop_fn *new_loop_fn,
-    void *new_loop_user_data,
-    const struct aws_shutdown_callback_options *shutdown_options);
-
-/** Creates an event loop group, with clock, number of loops to manage, the function to call for creating a new
- * event loop, and also pins all loops to hw threads on the same cpu_group (e.g. NUMA nodes). Note:
- * If el_count exceeds the number of hw threads in the cpu_group it will be ignored on the assumption that if you
- * care about NUMA, you don't want hyper-threads doing your IO and you especially don't want IO on a different node.
- */
-AWS_IO_API
-struct aws_event_loop_group *aws_event_loop_group_new_pinned_to_cpu_group(
-    struct aws_allocator *alloc,
-    aws_io_clock_fn *clock,
-    uint16_t el_count,
-    uint16_t cpu_group,
-    aws_new_event_loop_fn *new_loop_fn,
-    void *new_loop_user_data,
-    const struct aws_shutdown_callback_options *shutdown_options);
-
-/**
- * Initializes an event loop group with platform defaults. If max_threads == 0, then the
- * loop count will be the number of available processors on the machine / 2 (to exclude hyper-threads).
- * Otherwise, max_threads will be the number of event loops in the group.
- */
-AWS_IO_API
-struct aws_event_loop_group *aws_event_loop_group_new_default(
-    struct aws_allocator *alloc,
-    uint16_t max_threads,
-    const struct aws_shutdown_callback_options *shutdown_options);
-
-/** Creates an event loop group, with clock, number of loops to manage, the function to call for creating a new
- * event loop, and also pins all loops to hw threads on the same cpu_group (e.g. NUMA nodes). Note:
- * If el_count exceeds the number of hw threads in the cpu_group it will be clamped to the number of hw threads
- * on the assumption that if you care about NUMA, you don't want hyper-threads doing your IO and you especially
- * don't want IO on a different node.
- *
- * If max_threads == 0, then the
- * loop count will be the number of available processors in the cpu_group / 2 (to exclude hyper-threads)
- */
-AWS_IO_API
-struct aws_event_loop_group *aws_event_loop_group_new_default_pinned_to_cpu_group(
-    struct aws_allocator *alloc,
-    uint16_t max_threads,
-    uint16_t cpu_group,
-    const struct aws_shutdown_callback_options *shutdown_options);
+    struct aws_allocator *allocator,
+    const struct aws_event_loop_group_options *options);
```

Comment:
This is a breaking change. Probably should keep all the old group_new() functions, but mark them deprecated.

Reply:
Can't we just document the behavior and update the constructor to only apply the behavior if cpu pinning is enabled?  I do not want to keep the old functions in any way shape or form.

Reply:
These are functions that MUST be called by any C code written against our libraries. Removing them is going to break any and all code written against aws-c-***. All tests. All samples. All 3rdparty code we don't know exists.

IMO: Marking them deprecated, and just having them call into the new function is going to be far less pain

Reply:
It would be a lot less work true.  I can do that I suppose.  I'll probably roll back to before I updated all the tests/usage.